src/metadata.mjs

```diff
@@ -27,23 +31,23 @@ const createMetadata = slugger => {
       name: undefined,
       depth: -1,
     },
-    stability: undefined,
     properties: {},
+    stability: undefined,
```

Comment:
This is not related to this PR, but we need to update the `type`, `text`, and `name` fields in the `HeadingMetadataEntry` interface in the `src/types.d.ts` (Those fields can be undefined)

Reply:
Technically, it should never happen; if it happens, there's a bug since a heading is what defines a section. I've only set it as undefined as a fallback to prevent the parser from failing in such scenarios.

Maybe we should fail it in those scenarios?

Reply:
I haven't yet come up with an error manager that can report errors on the source documents. The idea is that during the parser process, we "report" any oddities on the source files. (like eslint)

Feel free to open an issue regarding implementing said feature and point this out there